### DISCUSSION TITLE
Rebalance intensities of different coasters, notably the B&M Inverted coaster

### DISCUSSION THREAD

**Discussion by TheAtomTwister** [OpenRCT2/OpenRCT2#16496] [Feature requests]
There are a number of rides that I and many other users do not use often because the system that yields excitement, intensity, and nausea ratings for different coasters is not all that forgiving on some rides, and in some cases unrealistically so.  Coasters like the Smiler at Alton Towers would not be ridden by guests because the presence of 14 inversions would drive up the intensity well beyond 10.0 in most cases, making the having of such a ride not feasible unless maybe you delicately build it as a B&M Sitdown coaster, which brings me to my next point: the B&M Inverted coaster is known for being very smooth-running and fun to ride.  There are some B&M Inverts that I have built that have way higher intensity ratings than they should.  The Gs are modest, and if the coaster were in real life, there is no way it would be intense enough to where people would not want to ride it.  Here are some examples of Inverted coasters and their respective intensities compared side-by-side with both similar Sit-downs and identical sit-downs.   The first is a dueling pair, Twisted Coral and Rough Water.
![Six Flags over Texas 2022-01-24 02-33-06](https://user-images.githubusercontent.com/97222164/150763509-4cdea20e-117c-4360-bd8e-4dd11b5fe180.png)
![Six Flags over Texas 2022-01-24 03-23-01](https://user-images.githubusercontent.com/97222164/150763513-d874e0c9-69ed-44d2-929c-eed347772d3a.png)
![Six Flags over Texas 2022-01-24 03-24-36](https://user-images.githubusercontent.com/97222164/150763820-7eba3003-cb75-4589-a97b-eabeae44a48c.png)
The first image is the inverted coaster, the second is sit-down, the third is floorless.  Neither of those inverts with those G-forces should be going over 10 in intensity.  They should be more like a mid 8 or low 9 at most.

As you can see, not all the track pieces have upright B&M sprites, notably the helixes.  We'll get to the inline twist previously discussed  down the road.  The difference in intensity is clear.  Why the inverts are so much more intense I think has a bit to do with RCT1 and intensity factors usually used for Vekoma rides being applied to the B&M inverts and those numbers were carried to RCT2 when numbers more similar to what the Floorless coaster would be more appropriate.  I believe a great deal of what went through the minds of the RCT2 developers when setting intensities were what the experiences of the rides were actually like.  Arrow rides tend to be a bit rougher than B&M rides, for example.

I do think that the intensity factor that comes from the number of inversions should be adjusted and the number of inversions should affect nausea more than intensity.  Here's some more examples, this one with a Standup B&M (which rightfully so should be quite a bit more intense than the sit-down having been on one myself.
![Six Flags over Texas 2022-01-24 02-33-41](https://user-images.githubusercontent.com/97222164/150765203-2a51f94b-b187-4f7b-8ca9-d34e68225c42.png)
![Six Flags over Texas 2022-01-24 02-59-36](https://user-images.githubusercontent.com/97222164/150765204-12aaff8e-9050-49f8-baa0-edd784903fac.png)
![Six Flags over Texas 2022-01-24 03-33-22](https://user-images.githubusercontent.com/97222164/150765206-b3e67eb7-9fe8-4856-85cf-f643c374cad3.png)
The first image is the Inverted, the second is Sit-down, the third is Stand-up.  The ratings for this are on Inverted Roller Coaster 1 and Twister Roller Coaster 1

This next one, Twisted Red Titanium, uses splicing from flying roller coasters to show what fully integrated inline twist track elements would work with when added to the B&M Twister track.  Unfortunately though, I do believe that the ride will perceive the flying roller coaster twists as adjacent rides and factor that in to the three ratings (excitement most notably).
![Six Flags over Texas 2022-01-24 02-34-42](https://user-images.githubusercontent.com/97222164/150765653-b5bde363-fe2d-4d69-8769-8d32b586ab3f.png)
![Six Flags over Texas 2022-01-24 03-17-54](https://user-images.githubusercontent.com/97222164/150765656-79cffc4e-b017-4a6b-80ce-340d37fa501a.png)

I definitely think that the B&M Inverted roller coaster's excitement, intensity, and nausea should be the sit-down's + 10-15% excitement, + 10% intensity, and + 10% nausea, but I'll leave the exact numbers up to you and would love to hear thoughts and opinions on what you think the ratings should be.  This by the way applies to all coasters, not just B&M ones.  I focus so much though on those because the ratings the Invert uses are so far off.  

Cheers and thanks to you all for putting so much work on such a fabulous upgrade to the game.  

P.S. I do have more ideas for tweaks and additions to share but will do so a little at a time so as to not spam the discussion threads.

**Comment by TheAtomTwister** [comment]
An alternate solution which I might like to see anyway is to add manual overrides for the excitement, intensity, and nausea ratings for rides and an enabler for manually overriding the ratings in the cheats menu.

**Reply by 733737** [reply]
the main issue is that this may break compatibility or effect new element competitions. You have to tread lightly on changing game mechanics

**Reply by TheAtomTwister** [reply]
Here I created in Paint.net an iteration of something that could be a way of implementing the mechanics that I'm talking about without interfering with any of the issues that you're talking about, and it took a bit of time to make this.

These are doctored versions of the same image of the dueling inverted coaster pair I used in the original post.  They showcase how the settings of which I speak could look if added to the game.  There can even be a function in the options menu that toggles the visibility of the settings in the stats tab.
![Six Flags over Texas 2022-01-24 02-33-06 adjustable ratings 1](https://user-images.githubusercontent.com/97222164/151171654-c32745d1-65c5-4a7e-b90b-60cda05eff39.png)
![Six Flags over Texas 2022-01-24 02-33-06 adjustable ratings 2](https://user-images.githubusercontent.com/97222164/151171656-cf40c436-7a0a-405e-9913-b9e886cc57eb.png)

As you can see, one image is what they look like with individual ride settings enabled, the other is when it is not.  The options menu setting would make the "vanilla", "realistic", and "manually set" options not visible at all.  The reason they would be visible but unchangeable with the ride rating override disabled is to show whether they're all set to "vanilla" or "realistic".

This function would allow for both sets of mechanics to co-exist and for people to toggle between them for either the whole park or for individual rides.  Only the coefficients for the excitement rating, intensity rating, and nausea rating would have a conditional change, and they would only be different if the "Set ratings to realistic" box is checked.  If the "Manually Set" box is checked, then they won't even be used for that ride.  This will allow for rides like the Smiler and other rides with ten or more inversions to exist in the park with an accurate trio of ratings, and guest's ridership of those rides would reflect those of their real-world counterparts.  It would be really good for park replicas.

Hope you find this helpful. 
 - TAT

**Reply by ZehMatt** [reply]
This has been probably suggested quite a few times over the years now but I don't think we'll be ever doing that, the amount of work required and extra code is not worth it.

**Reply by TheAtomTwister** [reply]
I might one day code it myself and give it to y'all if nobody does it first.

**Reply by pizza2004** [reply]
I think I might have suggested this before, but I think the best compromise *if* something like this is ever added is to make it scenario specific.  So people would need to manually choose what sort of logic is used in their scenarios and all older scenarios could default to the "classic" logic.  That way there is no risk of breaking old scenarios, but there's still the option to have stuff like this and someone could manually change the scenario settings in the debug menu.

That would also make it possible to load in RCT1 settings, RCT2 settings without a few slight bug fixes we've made, etc. if someone wanted to do that.

**Reply by TheAtomTwister** [reply]
That is noteworthy.  I would like to note that the "classic logic" in this instance would not be replaced by new logic, but rather the classic logic would remain and the new logic would be separate and independent of the classic logic so that one could toggle between the two ingame, as well as manually control the ratings in a way that is independent of both logics.  From my experience with coding Minecraft, I could probably figure out how to code all this in.  The only reason I don't do this right now is because I'd have to dedicate a significant amount of time and resources to preparing to and learning how to code this.  It is a project I'd like to do and I will probably wait until I decide to make seamless art for a large number of new track elements for Arrow and B&M track.  I would then code in the new logic when I add the art.

I am probably not going to code in the positioning of the coaster to the track though.  I would just provide the art to the OpenRCT2 community for future use.  Same goes for the "new logic", which would only be called upon when the proper setting in the cheats menu is active.  I am still split between ratings being set in "cheats" or in "options".

**Reply by ZehMatt** [reply]
We definitely do not want additional models calculating ride statistics, one has to do.

**Reply by TheAtomTwister** [reply]
Well in that case, I'd probably stick with the one we've got.  I do wish that the original developers didn't use the Vekoma numbers for B&M Inverts though, it's a bit unrealistic.  (I think they used the Vekoma numbers based on a dueling coaster I saw in the Swamp Cove scenario, but correct me if I'm wrong).

**Reply by 733737** [reply]
> Well in that case, I'd probably stick with the one we've got.  I do wish that the original developers didn't use the Vekoma numbers for B&M Inverts though, it's a bit unrealistic.  (I think they used the Vekoma numbers based on a dueling coaster I saw in the Swamp Cove scenario, but correct me if I'm wrong).

if you can prove the game has an original bug, the developers MAY consider swapping the existing 2 ridetypes' stats

**Reply by TheAtomTwister** [reply]
Here are the numbers of three Vekoma inverts (Compact Inverted Coaster) compared side-by-side with identical B&M inverts (Inverted Roller Coaster):
![Six Flags Great Adventure 2022-02-08 16-21-02](https://user-images.githubusercontent.com/97222164/153097532-bdfb0eff-18c8-4d48-8ace-cdc69b2d4ff6.png)
![Six Flags Great Adventure 2022-02-08 16-21-09](https://user-images.githubusercontent.com/97222164/153097535-60cefb8e-1d2a-4aea-bd96-7b12708af7c2.png)

There is clearly an increased excitement factor going from the Vekoma to the B&M.  If I were to make any changes at all to the ratings, it would be decreasing the intensity rating to what a Floorless counterpart would be.  There really is no bug here, and the numbers I would use come from real thrill ride experience.

It is possible that the OpenRCT2 community would respond positively to a change in intensity models for the inverted coaster, but I'm not sure that I would go through with something like that given the nature of what you're doing.  I would use it in an alternate model, and if you guys only want one model, I really would leave it alone.  Making a plugin with alternate ride stat models might be interesting though if anyone wants to do it.  I personally am a bit more into coding Minecraft than coding RCT2 (for now).

**Reply by TheAtomTwister** [reply]
I actually found an easy solution since the models seem to be built into the vehicles as I learned from the rides created by spacek531 and X123M3-256.  You could just create a duplicate Inverted coaster vehicle with different stats.  All the UI stuff isn't really necessary.

**Reply by ocalhoun6** [reply]
@TheAtomTwister 

You are a genius!

Seriously. The true mark of genius is simplicity, and this is a very elegantly simple solution to the otherwise unresolvable problem.

**Reply by TheAtomTwister** [reply]
Yeah, thinking more on it, you could just make vehicle clones of all of the train types that would have an alternate set of stats and have them labelled as such and then can decide which ones to use in the object selection.  The only reason I didn't think of that earlier was because I hadn't found all the ride vehicles released by those guys.  I've been using them and they really did a fantastic job on them.  I think the RMC ones are my favorite.  After finishing Poisoned Mines and then seeing the ratings on it it hit me, and I almost couldn't believe I didn't think of it before.  Here are the dueling coasters that inspired me: 
![Six Flags 2022-02-12 14-37-00](https://user-images.githubusercontent.com/97222164/153727575-9beb4e91-8b92-4e8e-a6b7-140b251448e1.png)

There's also Blood Lake in the background with X123M3-256's Intamin LSM trains and Gravity Gale with X123M3-256's 8-across B&M Dive trains.  I'm gonna be playing with these a lot now. ![veryhappyguest](https://user-images.githubusercontent.com/97222164/153727662-515c3bf9-d1e2-4e04-985f-81f12565e13a.png)
